src/main/java/stsjorbsmod/actions/PatronAction.java

```diff
@@ -0,0 +1,59 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.utility.WaitAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.vfx.combat.FlashAtkImgEffect;
+
+import static stsjorbsmod.util.CardMetaUtils.downgradePermanently;
+import static stsjorbsmod.util.CardMetaUtils.removeCard;
+import static stsjorbsmod.util.EffectUtils.showDestroyEffect;
+import static stsjorbsmod.util.EffectUtils.showDowngradeEffect;
+
+public class PatronAction extends AbstractGameAction {
+    private DamageInfo info;
+    private AbstractCard card;
+    private float baseDuration;
+
+    public PatronAction(AbstractCreature target, DamageInfo info, AbstractCard card) {
+        this.info = info;
+        this.setValues(target, info);
+        this.actionType = ActionType.DAMAGE;
+        this.duration = baseDuration = Settings.ACTION_DUR_MED;
+        this.card = card;
+    }
+
+    @Override
+    public void update() {
+        boolean isUpgraded = card.upgraded;
+        if (duration == baseDuration && this.target != null) {
+            AbstractDungeon.effectList.add(new FlashAtkImgEffect(this.target.hb.cX, this.target.hb.cY, AttackEffect.NONE));
+            // order here matters. We want to downgrade after damage happens, but remove before damage to tag with purgeOnUse for Wrath
+
+            if (isUpgraded) {
+                this.target.damage(this.info);
+                downgradePermanently(card, 0.25F);
+            } else {
+                removeCard(card);
+                this.target.damage(this.info);
+            }
+            if (AbstractDungeon.getCurrRoom().monsters.areMonstersBasicallyDead()) {
+                AbstractDungeon.actionManager.clearPostCombatActions();
+            }
+        }
+
+        tickDuration();
+        if (isDone) {
+            if (isUpgraded) {
+                showDowngradeEffect(card, duration);
```

Comment:
Is `duration` actually the field you mean to use here? It's the one `tickDuration` is decrementing, so it's always going to be negative at this point if `tickDuration` has set `isDone = true`

Reply:
This duration isn't even necessary.